templates/api/knowledge-panels/health/nutriscore/nutriscore.tt.json

```diff
@@ -24,7 +25,26 @@
                 }
             },
             [% END %]
-        [% END %]            
+        [% END %]
+        // Display actions if we are missing data to compute the Nutri-Score
+        [% IF panel.nutriscore_unknown_reason %]
+        {
+            "element_type": "action",
+            "action_element": {
+                "html": `[% lang("actions_to_compute_nutriscore") %]`,
+                "actions": [
+                    [% IF panel.nutriscore_unknown_reason == "missing_category" %]
+                        "add_categories"
+                    [% ELSIF panel.nutriscore_unknown_reason == "missing_nutrition_data" %]
+                        "edit_nutrition"
+                    [% ELSIF panel.nutriscore_unknown_reason == "missing_category_and_nutrition_data" %]
+                        "add_categories",
+                        "add_nutrition_facts"
+                    [% END %]
```

Comment:
Opinion: I think this kind of logic would be better on panel data computation. And then just have a list of actions.

Reply:
It's a possibility, but what I'm trying to do is to put the data generation logic in the Perl code, and the decisions on how to present that data to the user in the knowledge panels template. (but there are lots of panels where I wasn't really able to do that).

So for this case, the template currently generates a grouped list of actions, but the template could also separate them in different blocks, so that before each action we can explain why it's needed (e.g. why do we need a category for the Nutri-Score).

Reply:
ok